client/src/app/core/repositories/assignments/assignment-repository.service.ts

```diff
@@ -64,4 +71,142 @@ export class AssignmentRepositoryService extends BaseAgendaContentObjectReposito
         viewAssignment.getAgendaTitleWithType = () => this.getAgendaTitleWithType(viewAssignment);
         return viewAssignment;
     }
+
+    /**
+     * adds another user as a candidate
+     *
+     * @param candidate User id of a candidate
+     * @param assignment
+     */
+    public async addCandidate(candidate: number, assignment: ViewAssignment): Promise<void> {
+        const restPath = `/rest/assignments/assignment/${assignment.id}/candidature_other/`;
+        const data = { user: candidate };
+        await this.httpService.post(restPath, data);
+    }
+
+    /**
+     * Add the operator as candidate to the assignment
+     *
+     * @param assignment
+     */
+    public async addSelf(assignment: ViewAssignment): Promise<void> {
+        const restPath = `/rest/assignments/assignment/${assignment.id}/candidature_self/`;
+        await this.httpService.post(restPath);
+    }
+
+    public async addPoll(assignment: ViewAssignment): Promise<void> {
+        const restPath = `/rest/assignments/assignment/${assignment.id}/create_poll/`;
+        await this.httpService.post(restPath);
+        // TODO set phase, too, if phase was 0. Should be done server side?
+    }
+
+    /**
+     * Deletes a poll
+     * TODO: check if it is the correct one?
+     *
+     * @param id id of the poll to delete
+     */
+    public async deletePoll(id: number): Promise<void> {
+        const restPath = `/rest/assignments/poll/${id}/`;
+        await this.httpService.delete(restPath);
+    }
+
+    /**
+     * update data for a poll
+     *
+     * TODO documentation
+     * TODO check if votes is untouched
+     */
+    public async updatePoll(poll: Partial<Poll>, originalPoll: Poll): Promise<void> {
+        const restPath = `/rest/assignments/poll/${originalPoll.id}/`;
+        const data: Poll = Object.assign(originalPoll, poll);
+        await this.httpService.patch(restPath, data);
+    }
+
+    /**
+     * TODO: temporary update votes method. Needed because server needs different input than it's
+     * output in case of votes
+     *
+     * @param poll the updated Poll
+     *  @param originalPoll the original poll
+     */
+    public async updateVotes(poll: Partial<Poll>, originalPoll: Poll): Promise<void> {
```

Comment:
@FinnStutzenstein please help out here. (the stuff sent to the server is ugly (but as in 2.3); the server doesn't update the votes from the options; only votesvalid etc.)